src/OpenLoco/Windows/Construction/ConstructionTab.cpp

```diff
@@ -2472,26 +2475,113 @@ namespace OpenLoco::Ui::Windows::Construction::Construction
+            const auto westTileCoords = centreTileCoords - Map::offsets[1];
+            const auto northTileCoords = centreTileCoords + Map::offsets[3];
+            const auto southTileCoords = centreTileCoords - Map::offsets[3];
+
+            // Copy map elements which will be replaced with temporary ones containing track
+            backupTileElements[0] = *Map::TileManager::get(centreTileCoords)[0];
+            backupTileElements[1] = *Map::TileManager::get(eastTileCoords)[0];
+            backupTileElements[2] = *Map::TileManager::get(westTileCoords)[0];
+            backupTileElements[3] = *Map::TileManager::get(northTileCoords)[0];
+            backupTileElements[4] = *Map::TileManager::get(southTileCoords)[0];
+
+            // Set the temporary track element
+            Map::TrackElement newTrackEl(baseZ, clearZ, trackDirection, quarterTile /*.getBaseQuarterOccupied()*/, trackPiece.index, trackType, trackPieceId, std::nullopt, CompanyManager::getControllingId(), selectedMods);
+            newTrackEl.setLastFlag(true);
+
+            // Replace map elements with temp ones
+            *Map::TileManager::get(centreTileCoords)[0] = *reinterpret_cast<Map::TileElement*>(&newTrackEl);
+            *Map::TileManager::get(eastTileCoords)[0] = *reinterpret_cast<Map::TileElement*>(&previewSideSurfaceTileElement);
+            *Map::TileManager::get(westTileCoords)[0] = *reinterpret_cast<Map::TileElement*>(&previewSideSurfaceTileElement);
+            *Map::TileManager::get(northTileCoords)[0] = *reinterpret_cast<Map::TileElement*>(&previewSideSurfaceTileElement);
+            *Map::TileManager::get(southTileCoords)[0] = *reinterpret_cast<Map::TileElement*>(&previewSideSurfaceTileElement);
+
+            // Draw this map tile
+            Paint::paintTileElements(*session, trackPos);
+
+            // Restore map elements
+            *Map::TileManager::get(centreTileCoords)[0] = backupTileElements[0];
+            *Map::TileManager::get(eastTileCoords)[0] = backupTileElements[1];
+            *Map::TileManager::get(westTileCoords)[0] = backupTileElements[2];
+            *Map::TileManager::get(northTileCoords)[0] = backupTileElements[3];
+            *Map::TileManager::get(southTileCoords)[0] = backupTileElements[4];
+        }
+
+        session->arrangeStructs();
+        session->drawStructs();
+
+        Input::setMapSelectionFlags(backupSelectionFlags);
+        _constructionArrowPos = backupConstructionArrowPos;
+        _constructionArrowDirection = backupConstructionArrowDir;
+
         registers regs;
-        regs.ax = x;
-        regs.cx = y;
-        regs.edi = selectedMods << 16 | di;
-        regs.bh = bh;
-        regs.edx = colour << 16 | trackPieceId << 8 | trackType;
+        regs.ax = pos.x;
+        regs.cx = pos.y;
+        regs.edi = selectedMods << 16 | pos.z;
+        regs.bh = direction;
+        regs.edx = trackPieceId << 8 | trackType;
         call(0x004A0AE5, regs);
```

Comment:
Is this call left in for debugging purposes?

Reply:
oh doh i thought i deleted that.